openvdb_houdini/SOP_OpenVDB_Activate.cc

```diff
@@ -0,0 +1,604 @@
+
+        openvdb::CoordBBox    getIndexSpaceBounds(OP_Context &context,
+                                                  const GEO_PrimVDB &vdb);
+        UT_BoundingBox        getWorldBBox(fpreal t);
+
+    protected:
+        REGIONTYPE_NAMES REGIONTYPE(double t) { return (REGIONTYPE_NAMES) evalInt("regiontype", 0, t); }
+
+        OPERATION_NAMES OPERATION(fpreal t) { return (OPERATION_NAMES) evalInt("operation", 0, t); }
+
+
+        UT_Vector3D CENTER(fpreal t)
+            { return UT_Vector3D(evalFloat("center", 0, t),
+                                 evalFloat("center", 1, t),
+                                 evalFloat("center", 2, t)); }
+        UT_Vector3D SIZE(fpreal t)
+            { return UT_Vector3D(evalFloat("size", 0, t),
+                                 evalFloat("size", 1, t),
+                                 evalFloat("size", 2, t)); }
+
+        openvdb::Coord MINPOS(fpreal t)
+            { return openvdb::Coord(evalVec3i("min", t)); }
+        openvdb::Coord MAXPOS(fpreal t)
+            { return openvdb::Coord(evalVec3i("max", t)); }
+    };
+protected:
+             SOP_VDBActivate(OP_Network *net, const char *name, OP_Operator *entry);
+    ~SOP_VDBActivate() override {}
+
+    REGIONTYPE_NAMES REGIONTYPE(double t) { return (REGIONTYPE_NAMES) evalInt("regiontype", 0, t); }
+
+    OPERATION_NAMES OPERATION(fpreal t) { return (OPERATION_NAMES) evalInt("operation", 0, t); }
+};
+
+void
+#ifdef SESI_OPENVDB
+new_SOP_VDBActivate(OP_OperatorTable *table)
+#else
+newSopOperator(OP_OperatorTable *table)
+#endif
+{
+    if (table == NULL) return;
+
+    hutil::ParmList parms;
+
+    parms.add(hutil::ParmFactory(PRM_STRING, "group", "Source Group")
+              .setHelpText("Specify a subset of the input VDB grids to be processed.")
+              .setChoiceList(&hutil::PrimGroupMenuInput1));
+
+    // Match OPERATION
+    const char* operations[] = {
+        "union",        "Union",
+        "intersect",    "Intersect",
+        "subtract",     "A - B",
+        "copy",         "Copy",
+        NULL
+    };
+    parms.add(hutil::ParmFactory(PRM_ORD, "operation", "Operation")
+              .setDefault(PRMzeroDefaults)
+              .setChoiceListItems(PRM_CHOICELIST_SINGLE, operations));
+
+    parms.add(hutil::ParmFactory(PRM_TOGGLE, "setvalue", "Write Value")
+                .setTypeExtended(PRM_TYPE_TOGGLE_JOIN)
+                .setDefault(PRMoneDefaults));
+    parms.add(hutil::ParmFactory(PRM_FLT, "value", "Value")
+                .setDefault(PRMoneDefaults));
+
+    // Match REGIONTYPE
+    std::vector<PRM_Default> items;
+    items.push_back(PRM_Default(2, "Position"));
+    items.push_back(PRM_Default(2, "Voxel"));
+    items.push_back(PRM_Default(1, "Expand"));
+    items.push_back(PRM_Default(2, "Reference"));
+    items.push_back(PRM_Default(0, "Deactivate"));
+    items.push_back(PRM_Default(0, "Fill SDF"));
+    parms.add(hutil::ParmFactory(PRM_SWITCHER_EXCLUSIVE, "regiontype", "Region Type")
```

Comment:
Yeah, in practice no one should ever use the exclusive switcher as it is bad UX.  But it is what it is :>

Reply:
I've switched this to the beginExclusiveSwitcher.  Much nicer and less error prone.  Still ideally it would be a drop-down menu and a bunch of hideWhens.

Reply:
Thanks, that's much clearer!